core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/replication-repmgr-Debian.yml

```diff
@@ -0,0 +1,174 @@
+  postgresql_user:
+    user: "{{ specification.additional_components.replication.user }}"
+    password: "{{ specification.additional_components.replication.password }}"
+    role_attr_flags: LOGIN,SUPERUSER
+
+- name: Add replication user to postgresql
+  become_user: postgres
+  postgresql_user:
+    user: "{{ specification.additional_components.replication.user }}"
+    password: "{{ specification.additional_components.replication.password }}"
+    role_attr_flags: LOGIN,REPLICATION
+
+- name: Add replication user to pg_hba.conf
+  lineinfile:
+    path: /etc/postgresql/10/main/pg_hba.conf
+    line: "host    replication     {{ specification.additional_components.replication.user }}     0.0.0.0/0            md5"
+
+- name: Create pgpass file
+  template:
+    src: pgpass-repmgr.j2
+    dest: /var/lib/postgresql/.pgpass
+    owner: postgres
+    group: postgres
+    mode: 0600
+
+- name: Set Node ID
+  set_fact:
+    node_id: "{{ (groups.postgresql.index(inventory_hostname) + 1) }}"
+
+- name: Create repmgr config file
+  template:
+    src: repmgr.conf.j2
+    dest: /etc/postgresql/10/main/repmgr.conf
+    owner: postgres
+    group: postgres
+    mode: 0600
+
+- name: Creates directory
+  file:
+    path: /dbbackup/{{ inventory_hostname }}/backup/
+    state: directory
+    owner: postgres
+    group: postgres
+
+- name: Insert postgresql configuration
+  blockinfile:
+    path: /etc/postgresql/10/main/postgresql.conf
+    marker: "# ANSIBLE MANAGED BLOCK"
+    insertafter: "# Add settings for extensions here"
+    block: |
+      archive_mode = on
+      archive_command = 'test ! -f /dbbackup/{{ inventory_hostname }}/backup/%f && gzip -c < %p > /dbbackup/{{ inventory_hostname }}/backup/%f'
+      shared_preload_libraries = 'repmgr'
+
+- name: Create a new database for repmgr
+  become_user: postgres
+  postgresql_db:
+    name: "{{ specification.additional_components.replication.repmgr_database }}"
+    owner: "{{ specification.additional_components.replication.user }}"
+
+# Master:
+- name: Restart postgresql service
+  service:
+    name: postgresql
+    state: restarted
+  when:
+    - groups['postgresql'][0] == inventory_hostname
+
+- name: Restart repmgrd service
+  service:
+    name: repmgrd
+    state: restarted
+  when:
+    - groups['postgresql'][0] == inventory_hostname
+
+- name: Register primary with repmgr
+  become_user: postgres
+  shell: "repmgr primary register -f /etc/postgresql/10/main/repmgr.conf -F -S postgres"
+  when:
+    - groups['postgresql'][0] == inventory_hostname
+
+
+# Slave:
+- name: Turn on hot_standby
+  replace:
+    path: /etc/postgresql/10/main/postgresql.conf
+    regexp: "^#hot_standby = on"
+    replace: "hot_standby = on"
+    backup: yes
+  when:
+    - groups['postgresql'][1] == inventory_hostname
+
+- name: Check if node is registered already in repmgr
+  become_user: postgres
+  shell: "repmgr cluster show -f /etc/postgresql/10/main/repmgr.conf | grep -i {{ inventory_hostname }}"
```

Comment:
Please add pipefail to prevent situation like below:
```
[ubuntu ~]$ repmgr | grep -i host
Command 'repmgr' not found, but can be installed with:

sudo apt install repmgr-common

[ubuntu ~]$ echo $?
1
```